shared_model/backend/protobuf/common_objects/amount.hpp

```diff
@@ -30,20 +30,13 @@ namespace shared_model {
       Amount(const iroha::protocol::Amount &proto_amount)
           : multiprecision_repr([this] {
               const auto offset = 64u;
-              std::vector<uint64_t> result;
-              result.push_back(proto_amount_.value().first());
-              result.push_back(proto_amount_.value().second());
-              result.push_back(proto_amount_.value().third());
-              result.push_back(proto_amount_.value().fourth());
-
-              // last concat not requires shift
-              return std::accumulate(result.begin(),
-                                     result.end() - 1,
-                                     boost::multiprecision::uint256_t(),
-                                     [&offset](auto &res, auto &num) {
-                                       return (res | num) << offset;
-                                     })
-                  | *(result.end() - 1);
+              auto times = 3u;
+              boost::multiprecision::uint256_t result;
+              result |= proto_amount_.value().first()  << offset * times--;
+              result |= proto_amount_.value().second() << offset * times--;
+              result |= proto_amount_.value().third()  << offset * times--;
```

Comment:
That's much better, thanks!
Btw, does the clang-format align `<<` operator?

Reply:
I didn't find such parameter.